Named Data Networking (NDN) Tools

MostAwesomeDude opened the pull request:
###### Motivation for this change

I want to play with NDN.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

I had to use versions from git because this stuff is so bleeding-edge.

MostAwesomeDude committed:
nfd: init at 0.5.1

And use fetchFromGitHub.

MostAwesomeDude committed:
ndn-tools: init at 0.4

ndn-tools: use fetchFromGitHub

MostAwesomeDude committed:
ndn-repo-ng: init at 05zv7i

repo-ng: Fix up name and pick a working revision.

And use fetchFromGitHub too.

MostAwesomeDude commented:
Hm. These actually do have releases; I'm going to fix that and rebase.

0xABAB commented on pkgs/development/libraries/ndn-cxx/default.nix:
```diff
@@ -1,30 +1,36 @@
+  # To do battle with the tests, add --with-tests to the configure above. ~ C.
+  doCheck = false;
+  checkPhase = ''
+    LDD_LIBRARY_PATH=build/ build/unit-tests
```
Don't you mean LD_LIBRARY_PATH?

MostAwesomeDude replied:
Yes. Fixed. The tests still don't pass, of course.

0xABAB commented on pkgs/servers/named-data/nfd/default.nix:
```diff
@@ -0,0 +1,38 @@
+  preConfigure = ''
+    patchShebangs waf
+    ./waf configure \
+      --boost-includes=${boost.dev}/include \
```
Needs quoting.

MostAwesomeDude replied:
Just this one, or in all of the configures?

0xABAB replied:
All locations.

0xABAB reviewed (changes requested):
Minor stuff. Thanks for your contribution.

Mic92 commented on pkgs/tools/networking/named-data/repo-ng/default.nix:
```diff
@@ -0,0 +1,42 @@
+  version = "2b7b83";
+in
+stdenv.mkDerivation {
+  name = "ndn-tools-0.1-${version}";
```
name is probably wrong.

MostAwesomeDude replied:
What a mistake! Thanks.

MostAwesomeDude committed:
ndn-cxx, nfd, ndn-tools, repo-ng: Fix up quotes in configurePhase.

MostAwesomeDude commented:
I've addressed all feedback and am ready for merge. `nfd` works, at least. I haven't gotten far enough to test the "ndnsec" toolset. `repo-ng` is cantankerous, which is unsurprising since it is still immature.

MostAwesomeDude committed:
ndn: Fix wrong outputs.

matthewbauer reviewed (changes requested):
Fix merge conflicts

mmahut commented:
Are there any updates on this pull request, please?

MostAwesomeDude committed:
ndn-cxx: 0.5.1 -> 0.6.6

Also bump nfd, ndn-tools, repo-ng, as they all seem very tied-together
in terms of versioning.

MostAwesomeDude commented:
Hi, I rebased the branch and also bumped the versions of all new packages. It seems that, at some point, a derivation based on my ndn-cxx work got merged, so now the branch's history is a bit messy. I can redo all of the commits, if desired. I haven't been poking at this stuff for a while, though.

MostAwesomeDude committed:
all-packages: Fix mergo.

sjmackenzie commented:
@matthewbauer I believe the merge conflicts were resolved. Can you merge please?

Mic92 commented on pkgs/servers/named-data/nfd/default.nix:
```diff
@@ -0,0 +1,32 @@
+  };
+  buildInputs = [ wafHook
+                  libpcap doxygen boost pkgconfig python pythonPackages.sphinx
+                  openssl ndn-cxx ];
```
```
nativeBuildInputs = [ wafHook doxygen pkgconfig python pythonPackages.sphinx ];
```

Mic92 commented on pkgs/tools/networking/named-data/ndn-tools/default.nix:
```diff
@@ -0,0 +1,28 @@
+    sha256 = "1sjhyhc9ww38vyf3qmjzh5c4c0jc43ypzjmpqwsbq8y5zzln0gfx";
+  };
+
+  buildInputs = [ wafHook libpcap openssl boost pkgconfig python pythonPackages.sphinx git ndn-cxx ];
```
```
nativeBuildInputs = [ wafHook pkgconfig python pythonPackages.sphinx git ];
```

btw. why is `git` required here?

Mic92 commented on pkgs/tools/networking/named-data/repo-ng/default.nix:
```diff
@@ -0,0 +1,34 @@
+    sha256 = "0pdinvq3cf458bmm2mxdlidrmmr38in1563ra8q0ch3wp45va2di";
+  };
+  buildInputs = [ wafHook libpcap openssl doxygen boost sqlite pkgconfig
+                  python pythonPackages.sphinx ndn-cxx ];
```
```
nativeBuildInputs = [ wafHook doxygen pkgconfig python pythonPackages.sphinx ];
```

Mic92 commented on pkgs/development/libraries/ndn-cxx/default.nix:
```diff
@@ -1,24 +1,37 @@
   };
+
   nativeBuildInputs = [ pkgconfig wafHook ];
   buildInputs = [ openssl doxygen boost sqlite python pythonPackages.sphinx];
```
also doxygen, python and sphinx should be in nativeBuildInputs

MostAwesomeDude committed:
ndn: Address review feedback.

roberth commented:
ndn-cxx has been updated to 0.7.1 in master. You could rebase `-i` and omit the update, then add yourself to the maintainers in a new commit.
